ADR/0016-integration-service-promotion-logic.md

```diff
@@ -0,0 +1,86 @@
+# 16. Promotion logic in the Integration Service
+
+Date Documented: 2023-02-08
+
+## Status
+
+In Review
+
+## Context
+
+Before the merge of HACBS & AppStudio, the AppStudio build-service created the
+ApplicationSnapshot and promoted the content to the user's lowest environment as soon as
+the build was completed and once the environment was built.
+The integration-service fulfilled the same role for HACBS in addition to running the 
+IntegrationTestScenario tests first before promoting.
+
+Here, we are documenting the decision to consolidate and move the logic at one place to avoid duplicacy
+on the actions.
+
+Note: This functionality has now been completely dropped from the build-service side after the merger.
+
+### What is promotion logic ?
+
+The movement of the container image values of components, between environments, along a prescribed 
+promotion graph (e.g. ‘dev’ env -> ‘staging’ env -> ‘prod’ env).
+
+Once the Snapshot is tested and verified successfully, its contents will then be deployed to the user's 
+lowest environments as per the information concluded from the user’s namespace. 
+
+The “lowest environments” are the Environments with no explicit parentEnvironment set.
+
+## Decision
+
+Consolidate the promotion logic for both HACBS & AppStudio and move it to the integration-service after 
+the merger of Stonesoup.
+
+![](../diagrams/ADR-0016/promotion-logic.jpg)
+
+Three steps to follow in the integration-service to validate and promote the content of the successful Snapshot. 
+It:
```

Comment:
Is the second line here an accident or did this thought get interrupted?

Reply:
It wasn't but seems weird now :) . Removed